lib/vector-buffers/src/variants/disk_v2/mod.rs

```diff
@@ -18,42 +19,93 @@
 //! - endianness of the files is based on the host system (we don't support loading the buffer files
 //!   on a system with different endianness)
 //!
-//! ## On-disk layout
+//! ## High-level design
+//!
+//! ### Records
+//!
+//! A record is an length-prefixed payload, where an arbitrary number of bytes are contained,
+//! alongside a monotonically increasing ID, and protected by a CRC32C checksum. Since a record
+//! simply stores opaque bytes, one or more events can be stored per record.
 //!
-//! At a high-level, records that are written end up in one of many underlying data files, while the
-//! ledger file -- number of records, writer and reader positions, etc -- is stored in a separate
-//! file. Data files function primarily with a "last process who touched it" ownership model: the
-//! writer always creates new files, and the reader deletes files when they have been fully read.
+//! The writer assigns record IDs based on number of events written to a record, such that a record
+//! ID of N can be determined to contain M-N events, where M is the record ID of the next record.
 //!
-//! ### Record structure
+//! #### On-disk format
 //!
-//! Records are packed together with a relatively simple pseudo-structure:
+//! Records are represented by the following pseudo-structure:
 //!
 //!   record:
 //!     `record_len`: uint64
 //!     `checksum`:   uint32(CRC32C of `record_id` + `payload`)
 //!     `record_id`:  uint64
-//!     `payload`:    uint8[]
+//!     `payload`:    uint8[record_len]
+//!
+//!
+//! We say "pseudo-structure" as a helper serialization library, `rkyv`, is used to handle
```

Comment:
Worth linking to the `rkyv` crate, or does that magically happen?

Reply:
Doesn't happen automatically, but I'll do that.

Reply:
Addressed in ce021e1a1.